File: files/creeps/src/main/java/com/epita/creeps/Creepstants.java
package com.epita.creeps;

/**
 * Here you will find all the necessary constants to run a proper IA of creeps.
 * All durations are given in ticks.
 *
 * To convert those constants into milliseconds, the formula is `(1000 * CONSTANT) / TICKRATE`
 *
 * @author PSX
 * @since 1.0-SNAPSHOT
 */
public class Creepstants {

// Conf ---------------------------------------------------------------------------------------------------------------

    public static Integer TICKRATE = 20;
    public static Integer TIMEOUT = 30 * 1000;
    public static Integer ALLOWED_MISSES = 100;
    public static Integer PLAYER_START_BIOMASS = 200;
    public static Integer PLAYER_START_MINERALS = 200;

// Spawns -------------------------------------------------------------------------------------------------------------

    public static Integer PROBE_MINERAL = 50;
    public static Integer PROBE_BIOMASS = 5;
    public static Integer PROBE_SPAWNTIME = TICKRATE * 15;

    public static Integer SCOUT_MINERAL = 150;
    public static Integer SCOUT_BIOMASS = 0;
    public static Integer SCOUT_SPAWNTIME = TICKRATE * 5;

    public static Integer TEMPLAR_MINERAL = 100;
    public static Integer TEMPLAR_BIOMASS = 300;
    public static Integer TEMPLAR_SPAWNTIME = TICKRATE * 30;

    public static Integer BEACON_MINERAL = 200;
    public static Integer BEACON_BIOMASS = 500;
    public static Integer BEACON_SPAWNTIME = TICKRATE * 30;

    public static Integer NEXUS_MINERAL = 300;
    public static Integer NEXUS_BIOMASS = 300;
    public static Integer NEXUS_SPAWNTIME = TICKRATE * 15;

// Actions ------------------------------------------------------------------------------------------------------------

    public static Integer NOOP_TIME = TICKRATE * 1;
    public static Integer RELEASE_TIME = TICKRATE * 1;
    public static Integer DEFAULT_CONVERT_TIME = TICKRATE / 2;
    public static Integer DEFAULT_MINE_TIME = TICKRATE * 1;
    public static Integer DEFAULT_MOVE_TIME = TICKRATE / 2;
    public static Integer DEFAULT_PLAYERSTATUS_TIME = TICKRATE * 1;
    public static Integer DEFAULT_SCAN3_TIME = TICKRATE * 1;
    public static Integer DEFAULT_SCAN5_TIME = TICKRATE * 2;
    public static Integer DEFAULT_SCAN9_TIME = TICKRATE * 4;
    public static Integer DEFAULT_STATUS_TIME = TICKRATE / 2;
    public static Integer DEFAULT_SPHERE_TIME = TICKRATE * 10;
    public static Integer DEFAULT_IONDISCHARGE_TIME = TICKRATE * 10;
    public static Integer DEFAULT_ORBITALLASER_TIME = TICKRATE * 20;
}

File: server/sources/server/src/main/kotlin/com/mti/creeps/Creepstants.kt
package com.mti.creeps

/**
 * You may want to add a meaningful comment here.
 *
 * @author PSX
 * @since 1.0-SNAPSHOT
 */

// Conf ---------------------------------------------------------------------------------------------------------------

val TICKRATE: Int = 20;
val TIMEOUT: Int = 30 * 1000;
val ALLOWED_MISSES: Int = 100
val PLAYER_START_BIOMASS: Int = 200
val PLAYER_START_MINERALS: Int = 200


// Spawns -------------------------------------------------------------------------------------------------------------

val PROBE_MINERAL: Int = 50;
val PROBE_BIOMASS: Int = 5;
val PROBE_SPAWNTIME: Int = TICKRATE * 15;
val PROBE_SPAWNTIME_READYUP: Int = TICKRATE * 0;

val SCOUT_MINERAL: Int = 150;
val SCOUT_BIOMASS: Int = 0;
val SCOUT_SPAWNTIME: Int = TICKRATE * 5;
val SCOUT_SPAWNTIME_READYUP: Int = TICKRATE * 0;

val TEMPLAR_MINERAL: Int = 100;
val TEMPLAR_BIOMASS: Int = 300;
val TEMPLAR_SPAWNTIME: Int = TICKRATE * 30;
val TEMPLAR_SPAWNTIME_READYUP: Int = TICKRATE * 0;

val BEACON_MINERAL: Int = 200;
val BEACON_BIOMASS: Int = 500;
val BEACON_SPAWNTIME: Int = TICKRATE * 30;
val BEACON_SPAWNTIME_READYUP: Int = TICKRATE * 0;

val NEXUS_MINERAL: Int = 300;
val NEXUS_BIOMASS: Int = 300;
val NEXUS_SPAWNTIME: Int = TICKRATE * 15;
val NEXUS_SPAWNTIME_READYUP: Int = TICKRATE * 0;

val PYLON_MINERAL: Int = 100;
val PYLON_BIOMASS: Int = 50;
val PYLON_SPAWNTIME: Int = TICKRATE * 8;
val PYLON_SPAWNTIME_READYUP: Int = TICKRATE * 0;


// Actions ------------------------------------------------------------------------------------------------------------

val NOOP_POSTTIME: Int = TICKRATE * 0;
val NOOP_PRETIME: Int = TICKRATE * 1;

val RELEASE_POSTTIME: Int = TICKRATE * 0;
val RELEASE_PRETIME: Int = TICKRATE * 1;

val DEFAULT_CONVERT_POSTTIME: Int = TICKRATE * 0;
val DEFAULT_CONVERT_PRETIME: Int = TICKRATE / 2;

val DEFAULT_MINE_POSTTIME: Int = TICKRATE * 0;
val DEFAULT_MINE_PRETIME: Int = TICKRATE * 1;

val DEFAULT_MOVE_POSTTIME: Int = TICKRATE * 0;
val DEFAULT_MOVE_PRETIME: Int = TICKRATE / 2;

val DEFAULT_PLAYERSTATUS_POSTTIME: Int = TICKRATE / 2;
val DEFAULT_PLAYERSTATUS_PRETIME: Int = TICKRATE / 2;

val DEFAULT_SCAN3_POSTTIME: Int = TICKRATE / 2;
val DEFAULT_SCAN3_PRETIME: Int = TICKRATE / 2;

val DEFAULT_SCAN5_POSTTIME: Int = TICKRATE;
val DEFAULT_SCAN5_PRETIME: Int = TICKRATE;

val DEFAULT_SCAN9_POSTTIME: Int = TICKRATE * 2;
val DEFAULT_SCAN9_PRETIME: Int = TICKRATE * 2;

val DEFAULT_STATUS_POSTTIME: Int = TICKRATE / 4;
val DEFAULT_STATUS_PRETIME: Int = TICKRATE / 4;

val DEFAULT_SPHERE_POSTTIME: Int = TICKRATE * 5;
val DEFAULT_SPHERE_PRETIME: Int = TICKRATE * 5;

val DEFAULT_IONDISCHARGE_POSTTIME: Int = TICKRATE * 5;
val DEFAULT_IONDISCHARGE_PRETIME: Int = TICKRATE * 5;

val DEFAULT_ORBITALLASER_POSTTIME: Int = TICKRATE * 10;
val DEFAULT_ORBITALLASER_PRETIME: Int = TICKRATE * 10;